provider/cmd/pulumi-resource-nuage/nuage_provider/container_function.py

```diff
@@ -114,21 +116,52 @@ def __init__(self, name: str, args: ContainerFunctionArgs, props: Optional[dict]
         
         if args.repository:
             # Get existing repository using repository name.
-            repository = aws.ecr.get_repository(name=args.repository).repository_url
+            repository_url = aws.ecr.get_repository(name=args.repository).repository_url
         else:
```

Comment:
It's kind of messy to allow either the repository name or a Repository directly (aws.ecr.Repository or awsx.ecr.Repository... ? they don't have the same inputs)

How about simplifying to just `repository_id` ? This way if it's empty we know we need to create one !

Reply:
As for the name of the newly created repo, we can simply use the `name` of the ContainerFunction

BTW I suggest renaming `name` to `resource_name` --> It's more aligned with other Pulumi resources